### ISSUE TITLE
[MPD] owntone does not sanitize line feeds in tags in MPD response

### ISSUE THREAD

**Issue by skrobul** [owntone/owntone-server#1613]
When an audio file contains a line feed (hex 0a) in a tag this is retained in the MPD response for this file which is then correctly identified as an [Invalid response](https://github.com/ncmpcpp/ncmpcpp/issues/483) by clients.The linked example is for mopidy, but owntone behaves exactly the same.

Steps to reproduce and/or logging of the error:

1. Create an mp3 file with the line feed in the name. For example set `Artist` field to `Someone\nARTIST=Abc`
2. Add file to the library
3. Attempt to connect with one of the mpd clients. I tested with `ncmpc`, `ncmpcpp` which crashed with segfault, `mpc list artist` which crashed upon reaching the offending artist

- version of owntone: 28.5
- platform: x86_64

Screenshot from a tcpdump:

![image](https://github.com/owntone/owntone-server/assets/183976/d53838f5-1512-42a9-aba6-3c6d1df29df8)

**Comment by skrobul** [comment]
[example public domain mp3 file](https://github.com/owntone/owntone-server/files/11549165/example_mp3.zip) to reproduce the problem

```
$ exiftool -Artist 1937-xx-xx_BBC_Winston_Churchill_The_German_Ambassador.mp3
Artist                          : 1937-xx-xx BBC Winston Churchill.ARTIST=owntone_bug_demo


$ exiftool -Artist 1937-xx-xx_BBC_Winston_Churchill_The_German_Ambassador.mp3 -Ec
Artist                          : 1937-xx-xx BBC Winston Churchill\nARTIST=owntone_bug_demo
```

**Comment by skrobul** [comment]
> Do you know what the expected behavior should be? 

I looked into the [ID3v2.4](https://github.com/id3/ID3v2.4/blob/master/id3v2.4.0-frames.txt) spec and looks like the newline characters are allowed only in Comments, Lyrics and Terms of use related tags, so they should have not been allowed to be placed in the MP3 in the first place, but here we are. 

The MPD protocol does not seem to specify what should happen to the newlines, it only [mentions Escaping String Values](https://mpd.readthedocs.io/en/latest/protocol.html#escaping-string-values), but the official _libmpdclient_ [escapes double quotes and backslash](https://github.com/MusicPlayerDaemon/libmpdclient/blob/6cc9ea739c033e36541b16b7cf31edc41f6694d9/src/quote.c#L37).

In other words, I do not have a strong opinion on which behaviour is correct to use here. If I was doing that for my personal use, I would like to have the list of artists in such case joined by a comma. However, others may expect other behavior.

My main goal would be to stop the MPD clients from crashing, regardless of how it is achieved. Even excluding affected songs from being included would be fine too.

**Comment by ejurgensen** [comment]
Ok, I will make something that replaces linefeed/carriage return with spaces

**Comment by skrobul** [comment]
> Ok, I will make something that replaces linefeed/carriage return with spaces

@ejurgensen  I took [a stab at it](https://github.com/owntone/owntone-server/pull/1615), but I have also not used C for 15+ years so please let me know or  close it if it's terrible ;-)

**Comment by ejurgensen** [comment]
Thanks, the PR looks fine (only small things: a comment with "why", the function should be static, and code style should match). I have to redo some other stuff in mpd.c, so I will include this change when I do that.

**Comment by skrobul** [comment]
Thank you, addressed - wasn't sure about the negative indent for the nested control flow statements but hopefully it's good.
